src/decoder/mod.rs

```diff
@@ -203,13 +203,13 @@ impl<R: Read> Decoder<R> {
     }
 }
 
-struct ReadDecoder<R: Read> {
+struct ReadDecoder<'a, R: Read> {
     reader: BufReader<R>,
-    decoder: StreamingDecoder,
+    decoder: StreamingDecoder<'a>,
     at_eof: bool,
 }
```

Comment:
The decoder doesn't need to have a lifetime parameter. It will _always_ allocate the chunks itself, i.e. the decoder always produces `Cow::Owned()`. Then the `Info` struct within it should use `'static` and not expose the parameter further.

Reply:
Indeed, that simplifies the signatures a bit. Thanks.